src/lib/utils/SynapseClient.ts

```diff
@@ -1797,6 +1803,78 @@ export const getRestrictionInformation = (
     BackendDestinationEnum.REPO_ENDPOINT,
   )
 }
+/**
+ * Returns a paginated result of access requirements associated for an entity
+ *
+ * See https://rest-docs.synapse.org/rest/GET/entity/id/accessRequirement.html
+ *
+ * @param {(string | undefined)} sessionToken token of user
+ * @param {string} id id of entity
+ * @param {number} [limit=50]
+ * @param {number} [offset=0]
+ * @returns {Promise<PaginatedResults<AccessRequirement>>}
+ */
+const getAccessRequirement = (
+  sessionToken: string | undefined,
+  id: string,
+  limit: number = 50,
+  offset: number = 0,
+): Promise<PaginatedResults<AccessRequirement>> => {
+  const url = `/entity/${id}/accessRequirement?limit=${limit}&offset=${offset}`
+  return doGet<PaginatedResults<AccessRequirement>>(
+    url,
+    sessionToken,
+    undefined,
+    BackendDestinationEnum.REPO_ENDPOINT,
+  )
+}
+
+/**
+ * Returns all the access requirements associated to an entity {id}, calling the
+ * paginated getAccessRequirement service until all results are returned.
+ *
+ * @param {(string | undefined)} sessionToken token of user
+ * @param {string} id id of entity to lookup
+ * @returns {Promise<Array<AccessRequirement>>}
+ */
+export const getAllAccessRequirements = async (
+  sessionToken: string | undefined,
+  id: string,
+): Promise<Array<AccessRequirement>> => {
+  let isMoreData = true
+  const accessRequirementResults = [] as AccessRequirement[]
+  const limit = 50
+  let offset = 0
+  while (isMoreData) {
+    const data = await getAccessRequirement(sessionToken, id, limit, offset)
+    accessRequirementResults.concat(data.results)
+    if (data.totalNumberOfResults < 50) {
```

Comment:
I don't think  `accessRequirementResults.concat(data.results)` modifies accessRequirementResults. I think it should.

Reply:
Whoops! Good call, that would've been a fun bug